src/Accounts/Accounts/Models/RMProfileClient.cs

```diff
@@ -168,38 +172,44 @@ public bool TryRemoveContext(IAzureContext context)
                 // (tenant is present and subscription is not provided)
                 if (!string.IsNullOrEmpty(tenantIdOrName))
                 {
-                    Guid tempGuid = Guid.Empty;
-                    if (!Guid.TryParse(tenantIdOrName, out tempGuid))
+                    IAccessToken token = null;
+                    try
                     {
-                        var tenants = ListAccountTenants(account, environment, password, promptBehavior, promptAction);
-                        var matchesName = tenants.Where(t => t.GetPropertyAsArray(AzureTenant.Property.Domains)
-                                                                                           .Contains(tenantIdOrName, StringComparer.InvariantCultureIgnoreCase));
-                        var homeTenants = matchesName.FirstOrDefault(t => t.IsHome);
-                        var tenant = homeTenants ?? matchesName.FirstOrDefault();
-                        if (tenant == null || tenant.Id == null)
+                        token = AcquireAccessToken(
+                            account,
+                            environment,
+                            tenantIdOrName,
+                            password,
+                            promptBehavior,
+                            promptAction);
+
+                        if (!Guid.TryParse(tenantIdOrName, out Guid _))
                         {
-                            string baseMessage = string.Format(ProfileMessages.TenantDomainNotFound, tenantIdOrName);
-                            var typeMessageMap = new Dictionary<string, string>
+                            var tid = openIDConfigDoc.TenantId;
+                            token = new RawAccessToken()
+                            {
+                                AccessToken = token.AccessToken,
+                                LoginType = token.LoginType,
+                                TenantId = tid,
+                                UserId = token.UserId,
+                                HomeAccountId = token.HomeAccountId,
+                            };
+                            tenantIdOrName = token.TenantId;
```

Comment:
The tenantIdOrName looks like no need to be set again.

Reply:
tenantIdOrName is used after this setting.